client/signup/steps/jpo-summary/index.jsx

```diff
@@ -169,19 +209,57 @@ class JPOSummaryStep extends React.Component {
 					</tbody>
 				</table>
 				<div>
-					<Button primary onClick={ this.completeOnboarding }>
+					<a
+						className="jpo-summary__visit-site button is-primary"
+						href={ get( this.props.signupDependencies, [ 'jpoConnect', 'queryObject', 'home_url' ] ) }>
 						{
-							translate( 'Visit your site' )
+							translate( 'Visit site' )
 						}
-					</Button>
+					</a>
 				</div>
 			</div>
 		);
 	}
 
+	componentWillMount() {
+		let choices = this.getOnboardingChoices();
+		const siteId = get( this.props.signupDependencies, [ 'jpoConnect', 'queryObject', 'client_id' ], -1 ),
+			hasBusinessInfo = some( choices.onboarding.businessInfo, i => '' !== i );
+		if ( 'store' === choices.onboarding.genre ) {
+			this.props.installPlugin( siteId, { slug: 'woocommerce', id: 'woocommerce/woocommerce' } );
+		}
+		if ( hasBusinessInfo ) {
+			choices = extend( {}, choices, {
+				widgets: true,
+			} );
+		}
+		this.props.updateSettings(
+			siteId,
+			choices,
+			false // don't sanitize settings
+		).then( () => {
```

Comment:
Relying on this promise resolution is tricky as the promise will still resolve even if the API calls throws because we catch inside `updateSettings`. I understand that in this case there's no issue but it's probably a better pattern to do something like:
1. Calling `updateSettings()`. 
1. Reacting to a prop after it complets its job (maybe done inside this component's `connect()` statement)
1. Calling `addWidget` from `componentWillReceivProps` if we're in the state that we want after updatingSettings. 
1. Doing the same when `addWidget` finishes (i.e. reacting to a connected prop and showing or hiding what we want the user to see. 

This will allow you to still be in control of the component's state through a few props that indicate current state